### ISSUE TITLE
Support for forcing HTTP status code through HTML (for SPA)

### ISSUE THREAD

**Issue by rrjanbiah** [rendora/rendora#38]
Since, SPA handles 404 pages in the client side, it needs a mechanism to handle HTTP status code through HTML

For example, https://github.com/GoogleChrome/rendertron#status-codes provides below syntax to pass HTTP status code from HTML:
```html
<meta name="render:status_code" content="??" />
```

FWIW, tagging active fork authors in case if they have better solution implemented in their fork. @PureLandFlying @plan-my-binge @muhammadmuhlas @mehdipourfar

**Comment by muhammadmuhlas** [comment]
Hi @rrjanbiah 

It's a common case when returning 404 on SPA pages. You need to configure something like _redirects on netlify, or use try_files on nginx. You should see that on the deployment guide of your SPA.

**Comment by muhammadmuhlas** [comment]
Oh sorry, did you mean that rendora should look up on the meta tags about the http codes of the rendered pages instead of always returning 200?

**Comment by rrjanbiah** [comment]
@muhammadmuhlas 

> Oh sorry, did you mean that rendora should look up on the meta tags about the http codes of the rendered pages instead of always returning 200?

Yep, exactly. Similar to Rendertron https://github.com/GoogleChrome/rendertron#status-codes, it should support

```html
<meta name="render:status_code" content="??" />
```

**Comment by muhammadmuhlas** [comment]
It is possible to using strings.Contains on _get HTML Markup_ after _wait after DOM load_

**Comment by rrjanbiah** [comment]
> It is possible to using strings.Contains on _get HTML Markup_ after _wait after DOM load_

@muhammadmuhlas Thanks for your suggestion.

**Comment by muhammadmuhlas** [comment]
@rrjanbiah you could also use regex to get the code, but becareful using regex on html response, the cost could be expensive.

**Comment by rrjanbiah** [comment]
> @rrjanbiah you could also use regex to get the code, but becareful using regex on html response, the cost could be expensive.

Thank you so much for your help

**Comment by rrjanbiah** [comment]
*Side note: if anyone reached here through Google search*

1. Google seems to handle the 'Soft 404' correctly. Tested with g.co/mobile-friendly
2. For angular.io website, Google itself is handling 'Soft 404' it through `noindex` meta tag. As noted by https://stackoverflow.com/questions/35989950/how-to-notify-googlebot-about-404-pages-in-angular-spa/61700000#61700000
3. Yet, Rendertron like approach would be a cool feature for Rendora
